shared-module/displayio/Display.c

```diff
@@ -211,16 +212,25 @@ void displayio_display_end_transaction(displayio_display_obj_t* self) {
 }
 
 void displayio_display_set_region_to_update(displayio_display_obj_t* self, uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1) {
-    // TODO(tannewt): Handle displays with single byte bounds.
     uint16_t data[2];
     self->send(self->bus, true, &self->set_column_command, 1);
-    data[0] = __builtin_bswap16(x0 + self->colstart);
-    data[1] = __builtin_bswap16(x1 - 1 + self->colstart);
-    self->send(self->bus, false, (uint8_t*) data, 4);
+    if (self->single_byte_bounds) {
+        data[0] = __builtin_bswap16(((x0 + self->colstart) << 8) | ((x1 - 1 + self->colstart) & 0xFF));
```

Comment:
Instead of using bswap and shifts here, put the `data` declaration inside the if and change it to uint8_t. That will isolate the individual values and make it clearer to read.

Reply:
Good point on readability. I was just hoping it would make the program a little smaller when I did it this way. Thanks for merging it in. I can make the readability improvement for the next release.